File: DSA Self Paced/Arrays/Efficient Methods/CheckArraySorted.cpp
/**
 * @file CheckArraySorted.cpp
 * @author ADITYA BHARDWAJ
 * @brief 
 * @version 0.1
 * @date 2022-08-29 10:07PM
 * 
 * @copyright Copyright (c) 2022
 * 
 */

#include <bits/stdc++.h>

using namespace std;

bool checkArraySorted(vector<int> arr)
{
    
    for (int i = 1; i< arr.size(); i++)
        if (arr[i] < arr[i-1])
            return false;

    return true;
}

int main()
{
    vector<int> arr = {1, 2, 3, 4};
    // vector<int> arr = {1, 2, 5, 4};

    cout << checkArraySorted(arr);

    return 0;
}

File: DSA Self Paced/Arrays/Efficient Methods/CheckArraySorted.java
import java.util.ArrayList;

/**
 * @file CheckArraySorted.java
 * @author ADITYA BHARDWAJ
 * @brief 
 * @version 0.1
 * @date 2022-09-01 07:17PM
 * 
 * @copyright Copyright (c) 2022
 * 
 */

class CheckArraySorted{

    static boolean isSorted(ArrayList<Integer> arr)
    {
        for (int i=1; i< arr.size(); i++)
            if (arr.get(i) < arr.get(i-1))
                return false;
        
        return true;
    }

    public static void main(String[] args) {

        ArrayList<Integer> arr = new ArrayList<Integer>();
        // arr.add(8);
        // arr.add(12);
        // arr.add(15);

        arr.add(100);
        arr.add(20);
        arr.add(200);

        
        System.out.println(isSorted(arr));
    }
}